src/apdu_sign.c

```diff
@@ -770,6 +770,25 @@ static void slice_magic_bytes(buffer_t *buff) {
   else THROW(EXC_PARSE_ERROR);
 
 }
+
+static void replace_undisplayable(buffer_t *buff) {
+  uint8_t *data = buff -> bytes;
+  for(size_t i = 0; i < buff->length; i++) {
+    bool cant_display = data[i] > 126 || data[i] < 32;
+    if(cant_display)
+      data[i] = '*';
+  }
+}
+static void handle_long_message(buffer_t *buff) {
+  uint8_t *data = buff -> bytes;
```

Comment:
This operator, weirdly enough, doesn't use spaces around it, because field accessors generally don't (e.g. buff.bytes) and this is a type of field accessor.